src/services/cartItemService.ts

```diff
@@ -170,23 +212,41 @@ export const updateCartItemQuantityService = async( cartId: string, newQuantity:
   // updating the cart item's quantity
   await updateCartItemQuantity(cartItem.id, finalQuantity);
 
-  // fetching the updated cart item with the product details
-  const updatedCartItem = await getCartByCartId(cartItem.id);
+  // Fetch updated discount information
+  const [enrichedProduct] = await addCustomFields([product]);
```

Comment:
try -catch

Reply:
@batooldshilleh  can you check the file again? i added the try-catch to the update service